.dockerignore

```diff
@@ -0,0 +1,6 @@
+.dockerignore
+**.pyc
+**/__pycache__
+.gitignore
+.git
+./allennlp/data/fields/__pycache__
```

Comment:
I also question why we would need a separate line for this particular __pycache__